src/tools/rbd/action/Group.cc

```diff
@@ -760,20 +773,76 @@ int execute_group_snap_list(const po::variables_map &vm,
     t.define_column("NAME", TextTable::LEFT, TextTable::LEFT);
     t.define_column("STATE", TextTable::LEFT, TextTable::RIGHT);
+    t.define_column("NAMESPACE", TextTable::LEFT, TextTable::LEFT);
   }
 
   for (const auto& snap : snaps) {
-    auto state_string = get_group_snap_state_name(snap.state);
+    librbd::group_snap_mirror_namespace_t mirror_snap;
+    std::string mirror_snap_state = "unknown";
+    std::string snap_type = get_group_snap_namespace_name(snap.namespace_type);
+    std::string state_string = get_group_snap_state_name(snap.state);
+    if (snap.namespace_type == RBD_GROUP_SNAP_NAMESPACE_TYPE_MIRROR) {
+      r = rbd.group_snap_get_mirror_namespace(
+          io_ctx, group_name.c_str(), snap.id.c_str(), &mirror_snap);
+      if (r < 0) {
+        return r;
+      }
+      switch (mirror_snap.state) {
+        case RBD_SNAP_MIRROR_STATE_PRIMARY :
+            mirror_snap_state = "primary";
+            break;
+        case RBD_SNAP_MIRROR_STATE_NON_PRIMARY :
+            mirror_snap_state = "non-primary";
+            break;
+        case RBD_SNAP_MIRROR_STATE_PRIMARY_DEMOTED:
+        case RBD_SNAP_MIRROR_STATE_NON_PRIMARY_DEMOTED:
+            mirror_snap_state = "demoted";
+            break;
+      }
+    }
+
     if (f) {
       f->open_object_section("group_snap");
       f->dump_string("id", snap.id);
       f->dump_string("snapshot", snap.name);
       f->dump_string("state", state_string);
+      f->open_object_section("namespace");
+      f->dump_string("type", snap_type);
+      if (snap.namespace_type == RBD_GROUP_SNAP_NAMESPACE_TYPE_MIRROR) {
+        f->dump_string("state", mirror_snap_state);
+        f->open_array_section("mirror_peer_uuids");
+        for (auto &uuid : mirror_snap.mirror_peer_uuids) {
+          f->dump_string("peer_uuid", uuid);
+        }
+        f->close_section();
+        if (mirror_snap.state == RBD_SNAP_MIRROR_STATE_NON_PRIMARY ||
+            mirror_snap.state == RBD_SNAP_MIRROR_STATE_NON_PRIMARY_DEMOTED) {
+          f->dump_string("primary_mirror_uuid",
+                         mirror_snap.primary_mirror_uuid);
+          f->dump_string("primary_snap_id", mirror_snap.primary_snap_id);
+        }
+      }
+      f->close_section();
+      
       f->close_section();
     } else {
-      t << snap.id << snap.name << state_string << TextTable::endrow;
-    }
-  }
+      t << snap.id << snap.name << state_string;
+      std::ostringstream oss;
+      oss << snap_type;
+      if (snap.namespace_type == RBD_GROUP_SNAP_NAMESPACE_TYPE_MIRROR) {
+        oss << " (" << mirror_snap_state << " "
+                    << "peer_uuids:[" << mirror_snap.mirror_peer_uuids << "]";
+        if (mirror_snap.state == RBD_SNAP_MIRROR_STATE_NON_PRIMARY ||
+            mirror_snap.state == RBD_SNAP_MIRROR_STATE_NON_PRIMARY_DEMOTED) {
+          oss << " " << mirror_snap.primary_mirror_uuid << ":"
+              << mirror_snap.primary_snap_id;
```

Comment:
This doesn't seem right based on the output discussed in https://github.com/ceph/ceph/pull/62041#issuecomment-2737618262

Reply:
The output is made to look like the existing the one for  image snap listing.
See, `do_list_snaps` in `tools/rbd/action/Snap.cc`